crates/autopilot/src/run.rs

```diff
@@ -557,27 +561,41 @@ pub async fn run(args: Arguments) {
                 driver.name.clone(),
                 driver.fairness_threshold.map(Into::into),
                 driver.submission_account,
+                driver.accepts_unsettled_blocking,
             )
             .await
             .map(Arc::new)
             .expect("failed to load solver configuration")
         })
         .collect::<Vec<_>>();
 
-    let drivers = futures::future::join_all(drivers_futures)
+    let drivers: Vec<_> = futures::future::join_all(drivers_futures)
         .await
         .into_iter()
         .collect();
 
+    let solver_participation_guard = SolverParticipationGuard::new(
+        eth.clone(),
+        persistence.clone(),
+        competition_updates_receiver,
+        args.db_based_solver_participation_guard,
+        drivers
+            .iter()
+            .map(|driver| (driver.submission_address, driver.clone()))
+            .collect(),
```

Comment:
nit: storing the drivers in a map is an implementation detail the `Guard` cares about. Therefore it makes more sense to me for it to take an `impl IntoIterator<Item = Arc<Driver>>` and build the hashmap in the constructor. This would also avoid the possibility that the code to build the map has to be implemented multiple times.